can/io/asc.py

```diff
@@ -43,120 +43,156 @@ def __init__(self, file, base="hex"):
         """
         super().__init__(file, mode="r")
         self.base = base
+        self._converted_base = self._check_base(base)
+        self.date = None
+        self.timestamps_format = None
+        self.internal_events_logged = None
 
-    @staticmethod
-    def _extract_can_id(str_can_id, base):
+    def _extract_header(self):
+        for line in self.file:
+            line = line.strip()
+            lower_case = line.lower()
+            if lower_case.startswith("date"):
+                self.date = line[5:]
+            elif lower_case.startswith("base"):
+                try:
+                    _, base, _, timestamp_format = line.split()
+                except ValueError:
+                    raise Exception("Unsupported header string format: {}".format(line))
+                self.base = base
```

Comment:
The `__init__` takes a given `base`, whereas here right after that we're potentially overwriting it. What happens if the constructor specified `hex`, but the file is `dec`?

If you were intending to use this value from the `.asc` to validate, I don't think you should've overwritten `self.base` on this line.

Reply:
@karlding  In both the current and previous implementations, we were ignoring the base passed on to the constructor and using the base defined in the ASCII logs file to parse all the messages. It seems accurate to set base to the value actually used during parsing so I think overwriting it is fine.

Reply:
Hmm, if the idea is for the `ASCReader` to parse the value from the file, then we should probably just deprecate those parameters in `__init__` and raise a `DeprecationWarning` or something.

I don't have access to Vector's tools so I can't check, but are there cases where an ASC file is considered valid but it doesn't specify a base?

Reply:
Perhaps the only use case I can think of is when we have a logs file split across two ASCII files. Not completely sure if the header is present in files other than the first file. If the intention is perhaps to read only the second file, then having a manually specified base can be useful. Your call.

Reply:
Let's just leave this for now then. If someone has a use case for this, we can revisit it at that point.